core/src/main/java/org/infinispan/interceptors/SequentialInterceptorChain.java

```diff
@@ -0,0 +1,119 @@
+package org.infinispan.interceptors;
+
+import org.infinispan.commands.VisitableCommand;
+import org.infinispan.commons.util.Experimental;
+import org.infinispan.context.InvocationContext;
+import org.infinispan.interceptors.impl.InterceptorListNode;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Interceptor chain using {@link SequentialInterceptor}s.
+ *
+ * Experimental: The ability to modify the interceptors at runtime may be removed in future versions.
+ *
+ * @author Dan Berindei
+ * @since 8.2
+ */
+@Experimental
+public interface SequentialInterceptorChain {
+   List<SequentialInterceptor> getInterceptors();
```

Comment:
Missing some docs on these 2 methods.  Maybe change the name to be a bit more descriptive as well?